### ISSUE TITLE
disable rounded corners macos

### ISSUE THREAD

**Issue by jakenvac** [wezterm/wezterm#2182]
Hi, I've seen discussion about making the macos build preserve the rounded corners when the title bar is disabled here: #1034, I'm assuming the squared corners were originally considered a bug?

Is it possible to restore the squared corners in macos via a setting? If not, I think it would be a good idea to extend the window decorations setting to include a corners option.

**Comment by wez** [comment]
Yeah, all the windows are rounded on macOS. I'll accept a PR for something around this. I'm not super sure what the config should look like. Maybe add a `MACOS_SQUARE_CORNERS` to window decorations?

**Comment by rockyzhang24** [comment]
I think we can introduce `SQUARE_CORNER`. The existing flags `NONE`, `TITLE`, `RESIZE` and `TITLE | RESIZE` can be preserved, i.e., the rounded corner is the default. If users need the square corner, they can just append it to the corresponding flags. E.g., `NONE | SQUARE_CORNER`, `RESIZE | SQUARE_CORNER`, etc.

**Comment by wez** [comment]
An issue with just using `SQUARE_CORNER` is that it implies that not specifying it will give something other than square corners, but most systems only have square corners, and there is not a portable way to know or specify which should be used.

I think `MACOS_FORCE_SQUARE_CORNERS` is a more appropriately scoped name and effect.

**Comment by rockyzhang24** [comment]
Oh yes, exactly. `MACOS_FORCE_SQUARE_CORNERS` is proper flag. I second it.

**Comment by wez** [comment]
FWIW, I think this is more a matter of figuring out how to get macos to disable rounded corners with OpenGL and Metal views.

These are relevant questions with no obvious answer:

* https://stackoverflow.com/questions/38986010/when-exactly-does-an-nswindow-get-rounded-corners
* https://stackoverflow.com/questions/47173374/how-to-get-square-lower-corners-in-an-nswindow

https://github.com/lukakerr/NSWindowStyles#6-vibrant-background-with-custom-border-radius-and-no-titlebar might be adaptable for this purpose by setting the corner radius to 0.

I don't have time to actually sit down and play with this; I just happened to be looking at `window_decorations` tonight as part of #2669

**Comment by wez** [comment]
@karambaq not yet, but keep you eye on this PR:
* https://github.com/wez/wezterm/pull/2722

**Comment by IdoEsports** [comment]
hey @wez any update on the corner radius setting?

**Comment by dfl0** [comment]
Hey @wez, I was going to create an issue for this but noticed this one was asking exactly the same thing I was going to. Any plans to implement this at some point now? Is it a matter of difficulty or necessity?

**Comment by wez** [comment]
the PR-welcome tag indicates that I'm happy to review a PR produced by someone in the community, but that I have no plans to work directly on this myself.
